spec/compiler/semantic/automatic_cast_spec.cr

```diff
@@ -666,4 +666,16 @@ describe "Semantic: automatic cast" do
       foo(x)
       )) { float64 }
   end
+
+  it "can't autocast Int" do
```

Comment:
It's intentional though I forgot to change the "OH NO" to something else, just because the specs aren't working right now.

The idea is that `Int` can't be autocast to anything. We could make it autocast to `Int64`, but as @waj explained if later on `Int` can also be `Int128` on 128 bit platforms then that rule breaks. So it's better to disallow casting from `Int` into anything else.

However, `Int32` or smaller can be autocast to `Int` because I don't think we'll make `Int` be `Int16` in any platform.